### PR TITLE
Add resource openstack_lb_quota_v2

### LINKED ISSUE #1168: [New resource] Octavia quota
Octavia allows to set quotas per project. Add support for this resource.
This is part of [Gophercloud](https://github.com/gophercloud/gophercloud/tree/v0.15.0/openstack/loadbalancer/v2/quotas)
[Api docs](https://docs.openstack.org/api-ref/load-balancer/v2/?expanded=list-quota-detail#show-project-quota)

### PR DESCRIPTION
**Pull request by nikParasyr**
Close #1168 

Adds support for octavia quota.

In my understanding this cannot be implemented for neutron LBaaS

### CODE REVIEW

**Comment by nikParasyr** [comment]
Some notes
- Not sure if im handling microversions correctly. L7_rule and l7_policy were added in 2.19 as seen [here](https://github.com/openstack/octavia/blob/master/octavia/api/root_controller.py#L124). and also how to have acceptance tests it in different microversions (if that is required).
- In my understanding this is not implemented in neutron LBaas that is why i return error if not `use_octavia`
- I haven't added `region` as a attribute but maybe I should(?). If this is the case should I also append it in the resource id to avoid future issues such as https://github.com/terraform-provider-openstack/terraform-provider-openstack/issues/1104?

**Comment by jeblair** [comment]
recheck https://github.com/theopenlab/openlab/issues/681

**Comment by jeblair** [comment]
recheck
https://github.com/theopenlab/openlab/issues/681

**Comment by jeblair** [comment]
recheck

**Comment by nikParasyr** [comment]
recheck lbaas

**Comment by ozerovandrei** [comment]
> I haven't added region as a attribute but maybe I should(?). If this is the case should I also append it in the resource id to avoid future issues such as #1104?

I think that region is needed since there can be different quota for a single project in different regions.
Adding it in ID is fine. I think we can then do a refactoring of the existing quota resources in a similar way.

**Comment by nikParasyr** [comment]
@ozerovandrei Added region as an attribute and also appended on the resource ID.

File: website/docs/r/lb_quota_v2.html.markdown [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+layout: "openstack"
+page_title: "OpenStack: openstack_lb_quota_v2"
+sidebar_current: "docs-openstack-resource-lb-quota-v2"
+description: |-
+  Manages a V2 quota resource within OpenStack.
+---
+
+# openstack\_lb\_quota\_v2
+
+Manages a V3 block storage quota resource within OpenStack.
```
Comment (ozerovandrei) [inline website/docs/r/lb_quota_v2.html.markdown:11]: Not `V3 block storage quota` 😉

File: openstack/resource_openstack_lb_quota_v2.go [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+		_, err = quotas.Update(lbClient, projectID, updateOpts).Extract()
+		if err != nil {
+			return fmt.Errorf("Error updating openstack_lb_quota_v2: %s", err)
+		}
+	}
+
+	return resourceLoadBalancerQuotaV2Read(d, meta)
+}
+
+func parseLBQuotaID(id string) (string, string, error) {
```
Comment (ozerovandrei) [inline openstack/resource_openstack_lb_quota_v2.go:251]: Could you move this function to file with other helpers https://github.com/terraform-provider-openstack/terraform-provider-openstack/blob/main/openstack/lb_v2_shared.go and implement an unit test for it.

**Review by ozerovandrei — CHANGES_REQUESTED** [review]

File: website/docs/r/lb_quota_v2.html.markdown (hunk 1 above)
Comment (nikParasyr) [reply website/docs/r/lb_quota_v2.html.markdown]: :innocent: fixed :)

File: openstack/resource_openstack_lb_quota_v2.go (hunk 2 above)
Comment (nikParasyr) [reply openstack/resource_openstack_lb_quota_v2.go]: done as well

**Review by ozerovandrei — APPROVED** [review]
LGTM

**Pull request merged**